desc/optimize/least_squares.py

```diff
@@ -221,6 +221,27 @@ def lsqtr(  # noqa: C901 - FIXME: simplify this
 
         actual_reduction = -1
 
+        if max_nfev or maxiter == 0:
```

Comment:
do we need this if? or can we always just check termination before starting the loop to see if we're already at a local min?

Reply:
a few of the values needed for check_termination are not initially defined until the while loop, and the way the code is structured, it always check for termination since it will always enter the loop once. this just catches the case where with 0 maxiter, it never enters the loop and so never checks for termination